### DISCUSSION TITLE
Support for "Mariner3D" for SLA Printers

### DISCUSSION THREAD

**Discussion by NotExpectedYet** [OctoFarm/OctoFarm#225] [Feature Request]
Original Comment by: HyperToken

> Saw your post over at; https://github.com/luizribeiro/mariner
> 
> I've been using OctoFarm for all my FDM printers. I do realize there would be limited functionality, but this would be amazing to and my Mars Pro! I have Mariner all setup and working... let me know if you need a bug tester.

**Comment by NotExpectedYet** [comment]
Epic! And yes I own a mars Pro myself so will be getting it setup for testing. I'll keep you in mind as I will need bug testers. 

My plans from what I saw of the API we're to create a new printer definitions set for FDM, and then give this one Resin. I can also implement some Resin tracking similar to the Filament Manager in OctoFarm. I think it would also be beneficial to work these into the History and the Alerts. I see no issue with supporting all those currently. The UI needs some re-working but I've got some changes coming down in a month or two for that anyway.

**Reply by doucettom** [reply]
Just leaving a note here to follow-up as a flag to follow up ! Should be a resiner in a few days from now and willing to test stuff.

**Comment by Zenroth** [comment]
Just wanted to chime in and say I'm very interested in this, and would be willing to test it, and maybe even contribute some if my JS skills are up to the task. 

On the testing side, I currently have three printers I'm setting Mariner up on, and possibly a fourth though I need to open it up and see if its actually compatible with Mariner or not.

**Reply by NotExpectedYet** [reply]
We're setting this for 1.2 release of OctoFarm currently due to the current state of the management file. If you want to join in and help totally awesome. We're over at Discord with a developer-discussions channel if you'd like to join?

**Reply by doucettom** [reply]
My two Voxelab Proximas are now supported and I've set them up with Pi Cams so I can test stuff as well :)

**Reply by NotExpectedYet** [reply]
Sweet, I'm hoping to plan this in when I do the dreaded running refactor.

**Reply by NotExpectedYet** [reply]
Sweet, I'm hoping to plan this in when I do the dreaded running refactor.
